src/main/java/com/minecolonies/core/items/ItemAssistantHammer.java

```diff
@@ -0,0 +1,383 @@
+                .getSetting(BuildingMiner.FILL_BLOCK)
+                .getValue()
+                .getBlock()
+                .defaultBlockState());
+            handlers.add(0, solidPlaceHolderHandler);
+
+            final BuildAttemptResult buildAttemptResult = tryBuildingBlockNearby(player, colony, workOrder, interactPos, handlers);
+            if (buildAttemptResult.areBlocksToBuildNearby() && !buildAttemptResult.didTryBuilding())
+            {
+                player.displayClientMessage(Component.translatable("item.minecolonies.assistanthammer.noitems"), true);
+            }
+
+            if (buildAttemptResult.areBlocksToBuildNearby() && buildAttemptResult.didTryBuilding() && !player.isCreative())
+            {
+                player.getMainHandItem().hurtAndBreak(player.getMainHandItem().getItem().damageItem(player.getMainHandItem(), 1, player, s -> {}), player, s -> {});
+            }
+        }
+    }
+
+    @Override
+    public InteractionResult useOn(UseOnContext context)
+    {
+        if (context.getLevel().isClientSide)
+        {
+            final BlockPos interactPos = context.getClickedPos().relative(context.getClickedFace());
+            useOnBlock(context.getPlayer(), interactPos);
+        }
+
+        return InteractionResult.SUCCESS;
+    }
+
+    @NotNull
+    @Override
+    public InteractionResultHolder<ItemStack> use(Level level, Player player, InteractionHand hand)
+    {
+        if (level.isClientSide)
+        {
+            final BlockPos interactPos = BlockPos.containing(player.getEyePosition().add(player.getLookAngle().multiply(3, 3, 3)));
+            useOnBlock(player, interactPos);
+        }
+
+        return InteractionResultHolder.success(player.getMainHandItem());
+    }
+
+    /**
+     * Tries to build a block nearby, checking preview positions
+     *
+     * @param workOrder
+     * @param interactPos
+     * @param handlers
+     * @return
+     */
+    private @NotNull ItemAssistantHammer.BuildAttemptResult tryBuildingBlockNearby(
+        final Player player,
+        final IColony colony,
+        final IWorkOrder workOrder,
+        final BlockPos interactPos,
+        final List<IPlacementHandler> handlers)
+    {
+        final BlockPos.MutableBlockPos workPos = new BlockPos.MutableBlockPos();
+        boolean areBlocksToBuildNearby = false;
+        player.getCooldowns().addCooldown(this, 5);
+
+        for (int currentDistance = 0; currentDistance <= reach; currentDistance++)
+        {
+            for (int x = -currentDistance; x <= currentDistance; x++)
+            {
+                for (int y = -currentDistance; y <= currentDistance; y++)
+                {
+                    for (int z = -currentDistance; z <= currentDistance; z++)
+                    {
+                        int distanceSq = x * x + y * y + z * z;
+                        if (distanceSq > currentDistance * currentDistance)
+                        {
+                            continue;
```

Comment:
This feels inefficient. Shouldn't we have an early break as well?

Reply:
your feeling is off, we're only iterating 3-27 positions and just multiplication, its very fast. It returns on the first fitting position found

Reply:
kk